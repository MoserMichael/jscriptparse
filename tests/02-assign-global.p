

globVar = 1


def globNoChange(x) {
    globVar = x * x
    return globVar
}


def globVarChange(x) {
    globVar := x * x
    return globVar
}


println("globVar: {globVar}")
println("square:  {globNoChange(3)}")
println("globVar: {globVar} - not changed")
println("square:  {globVarChange(3)}")
println("globVar: {globVar} - did changed")





