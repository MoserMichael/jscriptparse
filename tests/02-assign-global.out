globVar: 1
square:  9
globVar: 1 - not changed
square:  9
globVar: 9 - did changed
